### ISSUE TITLE
Adding link on demo page is broken

### ISSUE THREAD

**Issue by ErneX** [jhollingworth/bootstrap-wysihtml5#48]
Trying to add a link on the demo page of the project throws this error:

TypeError: 'null' is not an object (evaluating 'attributes.text') wysihtml5-0.3.0_rc3.js:6912

Using Safari 5.1.5, OS X Lion 10.7.3

**Comment by naveenprakash** [comment]
Insert link is broken in Chrome(latest version), Mac osX

**Comment by ellisio** [comment]
This is still broken, any word on this?

**Comment by raldred** [comment]
Seems like it's something to do with the current selection in the editor, this is a pretty major issue if this doesn't work for me so I've forked to fix, will update with progress

**Comment by ErneX** [comment]
Yeah but the editor works fine if you implement it, somehow the demo is broken.

BTW: https://github.com/xing/wysihtml5 updated with support to links other than http (ftp, mailto, etc.).

**Comment by raldred** [comment]
Ahh I haven't actually implemented it yet, so I guess I would have found that out :)
Maybe the demo isn't using the latest source? No biggy then I guess.

**Comment by JoshLedgard** [comment]
We've implemented it and have the same issue.  To fix it we had to add a hidden div back to the toolbar code for the selection value to persist: 

```
                + "  <div data-wysihtml5-dialog='createLink' style='display: none;'>"
           + " <!--        <label>"
           + "           Link:"
           + "           <input data-wysihtml5-dialog-field='href' value='http://'>"
           + "         </label>"
           + "         <a data-wysihtml5-dialog-action='save'>OK</a>&nbsp;<a data-wysihtml5-dialog-action='cancel'>Cancel</a>-->"
            + "      </div>"
```

This seems like a total hack - but it works again.

**Comment by alekseyev** [comment]
Thank you @JoshLedgard, I also couldn't insert URLs in my installations without your hack
